java-connectors/kafka-connect-gcp-common/src/main/java/io/lenses/streamreactor/connect/gcp/common/auth/GCPConnectionConfig.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.gcp.common.auth;
+
+import io.lenses.streamreactor.common.config.base.RetryConfig;
+import io.lenses.streamreactor.common.config.base.intf.ConnectionConfig;
+import io.lenses.streamreactor.connect.gcp.common.auth.mode.AuthMode;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+import java.util.Optional;
+
+@Data
+@Builder
+@AllArgsConstructor
+public class GCPConnectionConfig  implements ConnectionConfig {
+
+    private Optional<String> projectId;
```

Comment:
This ties to my comments about validation in `GCPServiceBuilderConfigurer`. Many of those fields are `Optional`s which meant they're technically nullable. You want to leverage that using Lomboks `@Builder` but it's [not really intended to be used that way with Optionals](https://stackoverflow.com/questions/31670785/optional-in-lombok). This makes things complicated in your tests e.g. `GCPServiceBuilderConfigurerTest` where you're essentially setting all the things by yourself and those are Optionals.

My advice? See this article, the author was facing similar problem and thanks to the getters he avoided constructing the objects using `Optional`s